core/src/test/java/nu/yona/server/subscriptions/service/migration/MoveVpnPasswordToDeviceTest.java

```diff
@@ -0,0 +1,155 @@
+import nu.yona.server.subscriptions.entities.User;
+import nu.yona.server.subscriptions.entities.UserPrivate;
+import nu.yona.server.subscriptions.service.PrivateUserDataMigrationService.MigrationStep;
+import nu.yona.server.test.util.BaseSpringIntegrationTest;
+import nu.yona.server.test.util.CryptoSessionRule;
+import nu.yona.server.test.util.JUnitUtil;
+
+@Configuration
+@ComponentScan(useDefaultFilters = false, basePackages = { "nu.yona.server.subscriptions.service",
+		"nu.yona.server.device.service", "nu.yona.server.properties", "nu.yona.server" }, includeFilters = {
+				@ComponentScan.Filter(pattern = "nu.yona.server.subscriptions.service.migration.MoveVpnPasswordToDevice", type = FilterType.REGEX),
+				@ComponentScan.Filter(pattern = "nu.yona.server.device.service.DeviceService", type = FilterType.REGEX),
+				@ComponentScan.Filter(pattern = "nu.yona.server.Translator", type = FilterType.REGEX) })
+class MoveVpnPasswordToDeviceTestConfiguration extends UserRepositoriesConfiguration
+{
+	@Bean(name = "messageSource")
+	public ReloadableResourceBundleMessageSource messageSource()
+	{
+		return new CoreConfiguration().messageSource();
+	}
+
+	@Bean
+	UserDeviceRepository getMockUserDeviceRepository()
+	{
+		return new UserDeviceRepositoryMock();
+	}
+
+	@Bean
+	DeviceAnonymizedRepository getDeviceAnonymizedRepository()
+	{
+		return new DeviceAnonymizedRepositoryMock();
+	}
+
+	@Bean
+	BuddyAnonymizedRepository getBuddyAnonymizedRepository()
+	{
+		return new BuddyAnonymizedRepositoryMock();
+	}
+}
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = { MoveVpnPasswordToDeviceTestConfiguration.class })
+public class MoveVpnPasswordToDeviceTest extends BaseSpringIntegrationTest
+{
+	private static final String PASSWORD = "password";
+	private static final Field vpnPasswordField = JUnitUtil.getAccessibleField(UserPrivate.class, "vpnPassword");
+	private static final Field userPrivateField = JUnitUtil.getAccessibleField(User.class, "userPrivate");
+	private User richard;
+
+	@Autowired
+	private DeviceAnonymizedRepository deviceAnonymizedRepository;
+
+	@Autowired
+	private MigrationStep migrationStep;
+
+	@Rule
+	public MethodRule cryptoSession = new CryptoSessionRule(PASSWORD);
+
+	@Before
+	public void setUpPerTest() throws Exception
+	{
+		try (CryptoSession cryptoSession = CryptoSession.start(PASSWORD))
+		{
+			richard = JUnitUtil.createRichard();
+		}
+	}
+
+	@Override
+	protected Map<Class<?>, Repository<?, ?>> getRepositories()
+	{
+		Map<Class<?>, Repository<?, ?>> repositoriesMap = new HashMap<>();
+		repositoriesMap.put(DeviceAnonymized.class, deviceAnonymizedRepository);
+		return repositoriesMap;
+	}
+
+	@Test
+	public void upgrade_movePassword_passwordMoved() throws IllegalArgumentException, IllegalAccessException
+	{
+		String vpnPassword = "upgrade_movePassword_passwordMoved";
+		vpnPasswordField.set(userPrivateField.get(richard), vpnPassword);
+
```

Comment:
I would prefer to remove this newline and add one after the call to `upgrade` to separate arrange/act/assert.

Reply:
OK